telcell/data/utils.py

```diff
@@ -0,0 +1,62 @@
+import datetime
+from typing import Iterable, List, Tuple
+
+from telcell.data.models import Track
+
+
+def split_track_by_interval(
+        track: Track,
+        start: datetime.datetime,
+        end: datetime.datetime
+) -> Tuple[Track, Track]:
+    """
+    Splits the specified `track` in two separate `Track`s: one contains only
+    the `Measurement`s that fall within the specified [start, end) interval,
+    the other that contains all other measurements (i.e. those that occurred
+    before `start` or after `end`).
+
+    :param track: The track to be split in two
+    :param start: The start of the interval to split `track` on
+    :param end: The end of the interval to split `track` on (exclusive)
+    :return: Tracks of measurements within and outside the specified interval
+    """
+    selected_measurements = []
+    remaining_measurements = []
+
+    for measurement in track:
+        if start <= measurement.timestamp < end:
+            selected_measurements.append(measurement)
+        else:
+            remaining_measurements.append(measurement)
+
+    selected = Track(track.owner, track.name, selected_measurements)
+    remaining = Track(track.owner, track.name, remaining_measurements)
+    return selected, remaining
+
+
+def extract_intervals(
+        timestamps: Iterable[datetime.datetime],
+        start: datetime.datetime,
+        duration: datetime.timedelta,
+) -> List[Tuple[datetime.datetime, datetime.datetime]]:
```

Comment:
Voor beide functies vraag ik me af hoeveel pijn het volledig vullen van de verzamelingen gaat doen in een realistische setting. Zolang we niet willen kiezen of de data of de selectie aan modellen de 'outer loop' gaan vormen is dit vast logisch, maar mss een TODO waardig?